### PR TITLE
Various Hibernate ORM extension refactorings to pave the way for Hibernate Reactive

### PR DESCRIPTION
**Pull request by Sanne**
These are a set of simple (and rather safe looking?) changes I've extracted from a larger more complex PR.

I suspect having these CI verified (and merged?) already would make the overall review more manageable.

The additional upcoming fix is for #10187

### CODE REVIEW

**Comment by Sanne** [comment]
@gsmet let me add a bit of context now: sorry when I sent the PR I was in a big rush.

The gist of the fix for #10187 is that it needs to record its own custom `RecordedState` (as the metadata is different), so it needs to repeat most of the steps from the ORM processor.

The changes in this PR are mostly small code extractions and method extractions, so:
 - should be rather safer to apply
 - will allow the Hibernate Reactive extension to reuse those helpers so sharing the strategies

File: extensions/hibernate-orm/deployment/src/main/java/io/quarkus/hibernate/orm/deployment/HibernateOrmProcessor.java [hunk 1]
```diff
@@ -139,7 +138,6 @@
     private static final DotName STATIC_METAMODEL = DotName.createSimple(StaticMetamodel.class.getName());
 
     private static final String INTEGRATOR_SERVICE_FILE = "META-INF/services/org.hibernate.integrator.spi.Integrator";
-    private static final String SERVICE_CONTRIBUTOR_SERVICE_FILE = "META-INF/services/org.hibernate.service.spi.ServiceContributor";
```
Comment (gsmet) [inline extensions/hibernate-orm/deployment/src/main/java/io/quarkus/hibernate/orm/deployment/HibernateOrmProcessor.java:142]: Somehow, it looks like I missed something when I worked on that. We can probably remove it for now but at some point we might have to finish that properly.

**Review by gsmet — APPROVED** [review]
LGTM. Let's wait for CI.

File: extensions/hibernate-orm/deployment/src/main/java/io/quarkus/hibernate/orm/deployment/HibernateOrmProcessor.java (hunk 1 above)
Comment (Sanne) [reply extensions/hibernate-orm/deployment/src/main/java/io/quarkus/hibernate/orm/deployment/HibernateOrmProcessor.java:142]: yes, I also plan to rework how services are discovered, but need the Envers extension first.

**Comment by gsmet** [comment]
Merged, thanks!

**Comment by Sanne** [comment]
great, thanks.

**Pull request merged**